datalad_release_action/client.py

```diff
@@ -271,9 +261,7 @@ def ensure(self, label: Label) -> None:
                 log.info(
                     "Updating %s for label %r", ", ".join(payload.keys()), label.name
                 )
-                self.client.session.patch(
-                    f"{self.label_url}/{label.name}", json=payload
-                ).raise_for_status()
+                (self.label_ep / quote(label.name)).patch(json=payload)
```

Comment:
ha -- interesting pattern.  Is `ghreq.Endpoint` you came up with or you "borrowed" it from somewhere else?

Reply:
It's a descendant of code I wrote a long time ago.  I can't recall what the inspiration was at the time.